test/pdo_sqlsrv/pdo_fetch_variants_diff_styles.phpt

```diff
@@ -0,0 +1,296 @@
+    $stmt->bindColumn(20, $cols[19]);  
+    $stmt->bindColumn(21, $cols[20]);  
+    $stmt->bindColumn(22, $cols[21]);  
+    $stmt->bindColumn('c23_datetime2', $cols[22]);    
+
+    $stmt2 = $conn->query($query);
+    
+    // compare data values    
+    $row = 1;
+    while ($result = $stmt->fetch(PDO::FETCH_BOUND))
+    {   
+        echo "Comparing data in row $row\n";
+        
+        $obj = $stmt2->fetch(PDO::FETCH_LAZY);
+        if (! $obj)
+            echo "Failed to fetch data as object\n";
+        
+        $j = 0;
+        foreach ($cols as $value1)
+        {   
+            $col = $j+1;
+            $value2 = GetValueFromObject($obj, $col); 
+            DoValuesMatched($value1, $value2, $row, $col);
+            
+            $j++;
+        }
+        
+        $row++;
+    }
+    
+    $noActualRows = $row - 1;
+    if ($noActualRows != $numRows)
+    {
+        echo "Number of Actual Rows $noActualRows is unexpected!\n";
+    }
+        
+    $stmt = null;  
+    $stmt2 = null;   
+    
+    return $numCols;
+}
+
+function GetValueFromObject($obj, $col)
+{
+    switch ($col)
+    {
+        case 1: return $obj->c1_int;
+        case 2: return $obj->c2_tinyint;
+        case 3: return $obj->c3_smallint;
+        case 4: return $obj->c4_bigint;
+        case 5: return $obj->c5_bit;
+        case 6: return $obj->c6_float;
+        case 7: return $obj->c7_real;
+        case 8: return $obj->c8_decimal;
+        case 9: return $obj->c9_numeric;
+        case 10: return $obj->c10_money;
+        case 11: return $obj->c11_smallmoney;
+        case 12: return $obj->c12_char;
+        case 13: return $obj->c13_varchar;
+        case 14: return $obj->c14_nchar;
+        case 15: return $obj->c15_nvarchar;
+        case 16: return $obj->c16_binary;
+        case 17: return $obj->c17_varbinary;
+        case 18: return $obj->c18_uniqueidentifier;
+        case 19: return $obj->c19_datetime;
+        case 20: return $obj->c20_smalldatetime;
+        case 21: return $obj->c21_time;
+        case 22: return $obj->c22_date;
+        case 23: return $obj->c23_datetime2;
+        default: return null;
+    }    
+}
+
+function DoValuesMatched($value1, $value2, $row, $col)
+{
+    $matched = ($value1 === $value2);
+    if (! $matched)
+    {
+        echo "Values from row $row and column $col do not matched\n";
+        echo "One is $value1 but the other is $value2\n";
+    }    
+}
+
+function Fetch_Columns($conn, $tableName, $numRows, $numCols)
+{
+    try 
+    {
+        // insert column data from a row of the original table                    
+        $stmtOriginal = $conn->prepare("SELECT * FROM $tableName WHERE c1_int = :row");       
+        
+        for ($i = 1; $i <= $numRows; $i++)
+        {
+            $c1_int = $i;
+            
+            echo "Insert all columns from row $c1_int into one column of type sql_variant\n";            
+            $stmtOriginal->bindValue(':row', $c1_int, PDO::PARAM_INT);
+
```

Comment:
What if the user calls bindValue() without specifying the type? Will there be an error?